specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json

```diff
@@ -668,6 +668,50 @@
         }
       }
     },
+    "EncryptionDetails": {
+      "description": "Details of the encryption associated with the workspace",
+      "type": "object",
+      "properties": {
+        "doubleEncryptionEnabled": {
+          "description": "Double Encryption enabled",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "cmk": {
+          "description": "Customer Managed Key Details",
+          "$ref": "#/definitions/CustomerManagedKeyDetails"
+        }
+      }
+    },
+    "CustomerManagedKeyDetails": {
+      "description": "Details of the customer managed key associated with the workspace",
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "The customer managed key status on the workspace",
+          "type": "string",
+          "readOnly": true
+        },
+        "key": {
+          "description": "The key object of the workspace",
+          "$ref": "#/definitions/WorkspaceKeyDetails"
+        }
+      }
+    },
+    "WorkspaceKeyDetails": {
+      "description": "Details of the customer managed key associated with the workspace",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
You allow a PUT on the workspace referencing a child key that doesn't exist yet?  There isn't validation on this property?

Reply:
After the 2-step provisioning, we go ahead and only use the $.key.name as the key reference (no matter what URL is provided).